File: primary/controller/read/which_task.py
from hexagon.fvp.read.which_task import WhichTaskQuery
from primary.controller.dependency_list import DependencyList


def which_task(name: str, dependencies: DependencyList):
    set_of_open_tasks = dependencies.task_reader_for_fvp_which_task(name)
    set_of_fvp_sessions = dependencies.fvp_session_repository_for_fvp()

    response = WhichTaskQuery(set_of_open_tasks=set_of_open_tasks, set_of_fvp_sessions=set_of_fvp_sessions).which_task()
    return response

File: todolist-front-ts/primary/controller/whichTask.ts
import {WhichTask} from "@/hexagon/whichTaskQuery/whichTask.query";
import {StoreContract} from "@/primary/controller/chooseAndIgnoreTask";
import {Dependencies} from "@/primary/controller/dependencies";
import {WhichTaskUpdated} from "@/lib/todolist.slice";

export class Controller {
    private _whichTaskQuery: WhichTask.Contract;
    private _store: StoreContract;

    constructor(dependencies: Dependencies, store: StoreContract) {
        this._store = store;
        this._whichTaskQuery = dependencies.whichTask.useCase(dependencies)
    }


    async execute() {
        const tasksToExamine = await this._whichTaskQuery.query();

        this._store.dispatch(WhichTaskUpdated({
            tasks: tasksToExamine
        }));
    }

}